File: 89/89.java
class Solution {
    public List<Integer> grayCode(int n) {
        int nn = 1<<n;
        ArrayList<Integer> res = new ArrayList<Integer>(nn); 
        
        for (int i = 0; i < nn; i++){
            res.add( i^(i>>1) );
        }
        return res;
    }
}

File: 89/89.cs
public class Solution {
    public IList<int> GrayCode(int n) {
        int nn = 1<<n;
        // IList<int> res = new int[nn];
        IList<int> res = new List<int>();

        for (int i = 0; i < nn; i++){
            res.Add( i^(i>>1) );
        }
        return res;
    }
}